DOCS-2170 Document how to enable resource projects to a globalworkflow built from WDL (#516)
dnanexus/dxCompiler: doc/ExpertOptions.md

## Compiling Workflow

To compile a workflow:

```shell
java -jar dxCompiler-xxx.jar compile /path/to/foo.wdl -project project-xxxx -folder /my/workflows/
```

> This compiles `foo.wdl` to platform workflow `foo` in specified dx's project and folder (defaults to currently selected project and '/'). The generated workflow can then be run as usual using `dx run`. For example, if the workflow takes string argument `X`, then:

Reviewer: ```suggestion
This compiles `foo.wdl` to platform workflow `foo` in specified dx's project and folder (defaults the root directory of the current project). The generated workflow can then be run as usual using `dx run`. For example, if the workflow takes string argument `X`, then:
```

```shell

> dx run foo -i0.X="hello world"

Reviewer: ```suggestion
dx run foo -io.X="hello world"
```

# Publishing global workflows

A [global workflow](https://documentation.dnanexus.com/developer/workflows/version-and-publish-workflows#workflows-and-global-workflows) is an executable that can be versioned and published to other users. Publishing global workflows may facilitate collaboration across multiple projects, compared with local, project-based workflows.

Publishing a dxCompiler WDL workflow as a global workflow is supported from dxCompiler >= `v2.9.0` and dxpy >= `v0.319.2`. This is done in two steps. First, use `dxCompiler` to compile a workflow from WDL source to a local workflow in a project. Second, use `dx-toolkit` to publish the local workflow as a global workflow. Once the global workflow is published, you can add authorized users.

## Accessing Resource Projects and Files in Global Workflows

> When running as a global workflow, files and resources can be made available to all jobs and tasks by including them in the global workflow's resource container. To reference these files in your WDL script, use the environment variable `DX_GWF_RESOURCES_ID`. For compatibility with both global and local workflows, you can use the following pattern in your WDL command block:

Reviewer: ```suggestion
When running as a global workflow, files and resources can be made available to all jobs and tasks by including them in the global workflow's resource container. To reference the resource container ID with these files in your WDL script, use the environment variable `DX_GWF_RESOURCES_ID`. For compatibility with both global and local workflows, you can use the following pattern in your WDL command block:
```

```wdl
command <<<
  # Allow the docker container to evaluate the environmental variables
  source /home/dnanexus/environment

  # Determine which resource project to use
  if [[ "$DX_GWF_RESOURCES_ID" != "" ]]; then
    DX_ASSETS_ID="$DX_GWF_RESOURCES_ID"
  else

>     DX_ASSETS_ID="$DX_PROJECT_CONTEXT_ID"  # or a specific project ID

Reviewer: ```suggestion
    DX_ASSETS_ID="$DX_PROJECT_CONTEXT_ID"  # or a specific project ID which the workflow has access to
```
